### PR TITLE
AGM Location Change second PR

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#18161

*Description of changes:*
- Created the new AGM Location component
- Fixed validations
- Fixed spacing and sizing
- Added the component to the main framework
- Wired the component and data

**Upcoming PR: Unit tests + Update the "Request for AGM Location Change" in To Do in case payment fails? I don't know what I'll do there. I'll see.**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/AgmLocationChg.vue [hunk 1]
```diff
@@ -63,31 +63,29 @@
...
-                        <li>The company's articles permit a location outside BC;</li>
-                        <li>
-                          Nothing in the articles restrict a location change approved by
-                          resolution or by ordinary resolution, as the case may be.
-                        </li>
-                      </ul>
-                    </p>
-                  </div>
-                </section>
+                <h3 class="text-center">
```
Comment (JazzarKarim) [inline src/views/AgmLocationChg.vue:66]: Before:
before expendable help

After:
after expendable help

File: src/views/AgmLocationChg.vue [hunk 2]
```diff
@@ -416,6 +414,14 @@ export default class AgmLocationChg extends Mixins(CommonMixin, DateMixin,
     return (this.totalFee > 0)
   }
 
+  /** Array of validations rules for AGM location. */
+  get agmLocationRules (): Array<(val) => boolean | string> {
+    const rules = [] as Array<(val) => boolean | string>
+    rules.push(val => !!val || 'AGM location is required.')
+    rules.push(val => (val.length <= 100) || 'Must be less than 100 characters.')
```
Comment (JazzarKarim) [inline src/views/AgmLocationChg.vue:421]: Spoke to Yui about the rules here:
yui validation

File: src/views/AgmLocationChg.vue [hunk 3]
```diff
@@ -634,7 +632,8 @@ export default class AgmLocationChg extends Mixins(CommonMixin, DateMixin,
     const data: any = {
       [FilingTypes.AGM_LOCATION_CHANGE]: {
         year: this.agmYear,
-        newAgmLocation: this.agmLocationAddress
+        reason: this.reason,
+        agmLocation: this.agmLocation
```
Comment (JazzarKarim) [inline src/views/AgmLocationChg.vue:636]: Was looking at Hongjing's PR: https://github.com/bcgov/business-schemas/pull/146

Not sure if it'll change later but for now, looks like this is what we need.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-554-tkl68fqq.web.app

**Comment by severinbeauvais** [comment]
> **Upcoming PR: Unit tests + Update the "Request for AGM Location Change" in To Do in case payment fails? I don't know what I'll do there. I'll see.**

Check with Argus on what is needed to re-process the payment if it failed. Probably the user need to "retry payment" (or "change payment type"???).

File: src/views/AgmLocationChg.vue [hunk 4]
```diff
@@ -63,31 +63,29 @@
             <!-- Help -->
             <ExpandableHelp helpLabel="Help with Annual General Meeting Extension">
```
Comment (severinbeauvais) [inline src/views/AgmLocationChg.vue:64]: This is the wrong label. (It's wrong in the UI design, too.)

File: src/views/AgmLocationChg.vue [hunk 5]
```diff
@@ -63,31 +63,29 @@
...
+                      <li>The company's articles permit a location outside BC;</li>
+                      <li>
+                        Nothing in the articles restrict a location change approved by
+                        resolution or by ordinary resolution, as the case may be.
+                      </li>
+                    </ul>
+                  </p>
+                </div>
               </template>
             </ExpandableHelp>
```
Comment (severinbeauvais) [inline src/views/AgmLocationChg.vue:91]: Update line 97 below (see UI design).

File: src/views/AgmLocationChg.vue (hunk 2 above)
Comment (severinbeauvais) [inline src/views/AgmLocationChg.vue:421]: Error text is incorrect: 100 characters is allowed.

File: tests/unit/AgmLocationChg.spec.ts [hunk 6]
```diff
@@ -186,25 +186,25 @@ describe('AGM Location Chg view', () => {
     expect(!!vm.isPayRequired).toBe(true)
 
     // verify "validated" - all true
-    vm.agmLocationAddressValid = true
+    vm.agmLocationValid = true
```
Comment (severinbeauvais) [inline tests/unit/AgmLocationChg.spec.ts:189]: What about "reason valid"?

File: src/views/AgmLocationChg.vue [hunk 7]
```diff
@@ -402,7 +400,7 @@ export default class AgmLocationChg extends Mixins(CommonMixin, DateMixin,
 
   /** True if page is valid, else False. */
   get isPageValid (): boolean {
-    return (this.agmLocationAddressValid && this.agmYearValid && this.certifyFormValid)
+    return (this.agmLocationValid && this.agmYearValid && this.certifyFormValid)
```
Comment (severinbeauvais) [inline src/views/AgmLocationChg.vue:403]: What about "reason valid"?

**Review by severinbeauvais — COMMENTED** [review]
I noticed that I don't get an error dialog when I try to file (which fails due to API).

File: src/views/AgmLocationChg.vue (hunk 4 above)
Comment (severinbeauvais) [reply src/views/AgmLocationChg.vue]: Fixed, thanks.

File: src/views/AgmLocationChg.vue (hunk 5 above)
Comment (severinbeauvais) [reply src/views/AgmLocationChg.vue:91]: Fixed, thanks.

File: src/views/AgmLocationChg.vue (hunk 7 above)
Comment (severinbeauvais) [reply src/views/AgmLocationChg.vue]: Fixed, thanks.

File: tests/unit/AgmLocationChg.spec.ts (hunk 6 above)
Comment (severinbeauvais) [reply tests/unit/AgmLocationChg.spec.ts:189]: Fixed, thanks.

File: src/views/AgmLocationChg.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/views/AgmLocationChg.vue]: Fixed, thanks.

**Review by severinbeauvais — APPROVED** [review]
LGTM. Looking forward to next PR :)

PS I'm lovin' the small PRs.

**Comment by JazzarKarim** [comment]
> LGTM. Looking forward to next PR :)
> 
> PS I'm lovin' the small PRs.

Thanks Sev!

I also really really like them. I think it's always a great idea to go for smaller ones whenever we think a ticket's work is going to be big. It's much easier for the reviewers and the person working on it!

**Comment by JazzarKarim** [comment]
> > LGTM. Looking forward to next PR :)
> > PS I'm lovin' the small PRs.
> 
> Thanks Sev!
> 
> I also really really like them. I think it's always a great idea to go for smaller ones whenever we think a ticket's work is going to be big. It's much easier for the reviewers and the person working on it!

PS: There are two comments above from you Sev. I haven't forgot about them! I'll be dealing with them next PR 😎

**Review by leodube-aot — APPROVED** [review]
LGTM Karim!

**Pull request merged**